packages/react/label/src/Label.tsx

```diff
@@ -69,8 +53,37 @@ const Label = React.forwardRef<LabelElement, LabelProps>((props, forwardedRef) =
   }, [id, htmlFor]);
 
   return (
-    <LabelContext.Provider value={React.useMemo(() => ({ id, ref: labelRef }), [id])}>
-      <Primitive.span role="label" id={id} {...labelProps} ref={ref} />
+    <LabelContext.Provider
+      value={React.useMemo(() => ({ id, ref: labelRef, onControlChange: setControl }), [id])}
```

Comment:
```suggestion
      value={React.useMemo(() => ({ id, onControlChange: setControl }), [id])}
```

Looks like we don't need to pass this anymore, what i'm wondering though is why TS doesn't complain about any of the properties passed in here, am I missing something?

Reply:
Oh yeah, why didn't TS catch that 🤔

Reply:
Seems abstracting the context swallows the TS errors (including in latest TS versions) https://www.typescriptlang.org/play?ts=3.9.7#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdMAnmFnAGIQRwC8cA3nThwIAOywAuOKhhRgogOb1hMAO4QpogK4gARlij0AvvTq4xMuAGExMLAA94fbHhgA6XJWR2bou44AeTggAPgAKfiMUdGRRJgBKUwItUXxgMTgAeQBrZCYw+IEhOEoYLShROADff3cABWIAN2AAEwM4RuQAGy0sHn5+EXEpMjEsMgAaODUNOAAmKZgAC0pJcmXVsjgjKIB6ELojBmTUmHTKgCFibKxRAEYCouFzUUsX2t4BIbXR8Unp9RSBbTFZYH4bMFbEzFUrlSrVWwOepNVrtTo9Pr8d5IvYHI50E5pDJXCA3URzR6CYSwipVGpItwNCDNNpQDrdXr9Fz4NxaVBYACyWBAEDCjx4ITgEW+IzG-xmQMWoPByqh8SmAG0ALrxXGHY4pImXa63ADMlOKLzeiMcn257j5guFovFkulY1lf0WgPmStWIwh4226rg2sSMKwZVpCL8DKZLLRHMx2MceqMQA

Reply:
How annoying, as discussed earlier we could explicitly type the return / variable def to make it work but it hasn't been an issue anywhere else so probably fine tbh.

Reply:
Yeah, there are only a couple places we do this. Shall I make a separate PR to type these to avoid this in future?

cc @benoitgrelard

Reply:
>  Thoughts?

Agree, I'd prefer the type safety if it's not too much hassle for us to do so.

Reply:
I've done a quick test locally and that doesn't seem to fix it either 🙈 

![image](https://user-images.githubusercontent.com/175330/133419147-48e55961-ca7c-4d32-9395-59bbc5a57dd4.png)

the `one` and `two` properties should be erroring.

Reply:
Typing the return works for me

![image](https://user-images.githubusercontent.com/11708259/133420057-0fb85b4e-7100-4f51-8528-c08881679f61.png)

But as we're just talking about redundant properties I don't think it's a huge deal.

Reply:
ooooo nice!

Reply:
Proposed changes here https://github.com/radix-ui/primitives/pull/864

Reply:
Yeah it's not completely broken, it seems it's just not complaining about extra properties, but it will complain in all the cases about wrong values, or missing values.